ui/page/creatorMemberships/creatorArea/tiersTab/internal/editingTier/view.jsx

```diff
@@ -0,0 +1,207 @@
+    doMembershipList,
+  } = props;
+
+  const isMobile = useIsMobile();
+  const roughHeaderHeight = (isMobile ? 56 : 60) + 10; // @see: --header-height
+
+  const nameRef = React.useRef();
+  const contributionRef = React.useRef();
+
+  const [editTierParams, setEditTierParams] = React.useState({
+    editTierDescription: membership.Membership.description || '',
+    editTierName: membership.Membership.name || '',
+  });
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+
+  const nameError = getIsInputEmpty(editTierParams.editTierName);
+  const descriptionError = getIsInputEmpty(editTierParams.editTierDescription);
+
+  // custom emojis should be changed to channel member badge
+  const permanentTierPerks = ['Member badge'];
+
+  /**
+   * Check whether the tier already has the perk added
+   * @param {number} perkId - Id of the perk returned from DB
+   * @param {any} tier - Tier that is being edited
+   * @returns {boolean}
+   */
+  function containsPerk(perkId, tier) {
+    if (!tier.Perks) return false;
+
+    const perkIds = new Set(tier.Perks.map((tierPerk) => tierPerk.id));
+
+    return perkIds.has(perkId);
+  }
+
+  /**
+   * Selects the checked perks and builds them into a csv value for the backend
+   * @returns {string}
+   */
+  function generatePerksCsv() {
+    let selectedPerks = [];
+    for (const perk of membershipPerks) {
+      // $FlowFixMe
+      const odyseePerkSelected = document.querySelector(`input#perk_${perk.id}.membership_perks`).checked;
+      if (odyseePerkSelected) {
+        selectedPerks.push(perk.id);
+      }
+    }
+    return selectedPerks.toString();
+  }
+
+  /**
+   * When someone hits the 'Save' button from the edit functionality
+   * @param membershipTier - If an existing tier, use the old price and id
+   * @returns {Promise<void>}
+   */
+  async function saveMembership(membershipTier) {
+    setIsSubmitting(true);
+
+    const newTierMonthlyContribution = contributionRef.current?.input?.current?.value || 0;
+
+    const selectedPerksAsArray = generatePerksCsv();
+
+    if (activeChannelClaim) {
+      const isCreatingAMembership = typeof membershipTier.Membership.id === 'string';
+
+      doMembershipAddTier({
+        channel_name: activeChannelClaim.name,
+        channel_id: activeChannelClaim.claim_id,
+        name: editTierParams.editTierName,
+        description: editTierParams.editTierDescription,
+        amount: Number(newTierMonthlyContribution) * 100, // multiply to turn into cents
+        currency: 'usd', // hardcoded for now
+        perks: selectedPerksAsArray,
+        old_stripe_price: membershipTier.Prices ? membershipTier.Prices[0].id : undefined,
+        membership_id: isCreatingAMembership ? undefined : membershipTier.Membership.id,
+      }).then(() => {
+        setIsSubmitting(false);
+        removeEditing();
+        doMembershipList({ channel_name: activeChannelClaim.name, channel_id: activeChannelClaim.claim_id });
+      });
+    }
```

Comment:
:small_red_triangle: Handle error case for all API calls

In this case, I entered an invalid amount, and it just spins forever.

<img width="886" alt="image" src="https://user-images.githubusercontent.com/64950861/190604203-58f45bac-33d5-4628-bc57-85d12e0b7c2c.png">